packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -40,6 +40,44 @@ import 'theme_data.dart';
 ///
 /// The [clipBehavior] argument must not be null.
 ///
+/// {@tool sample}
+///
+/// This example shows how to use [FlatButton] to create a simple button
+/// in your flutter code.
+///
+/// ```dart
+/// FlatButton(
+///   onPressed: () {
+///     /*...*/
+///   },
+///   child: Text(
+///     "Flat Button",
+///     style: TextStyle(fontSize: 20.0),
+///   ),
+/// )
+/// ```
+/// This example shows how to use [FlatButton] to create a styled
+/// button in your flutter code.
```

Comment:
maybe describe what the style is? Something like:

"This example shows a button that is normally white-on-blue, with splashes rendered in a different shade of blue. It turns black-on-grey when disabled. The button has 8px of padding on each side, and the text is 20px high.